test/e2e/helpers.js

```diff
@@ -980,29 +973,18 @@ async function validateContractDetails(driver) {
   verifyContractDetailsButton.click();
   await driver.clickElement({ text: 'Got it', tag: 'button' });
 
-  // Approve signing typed data
-  try {
-    await driver.clickElement(
-      '[data-testid="signature-request-scroll-button"]',
-    );
-  } catch (error) {
-    // Ignore error if scroll button is not present
-  }
-  await driver.delay(regularDelayMs);
+  await driver.clickElementSafe(
+    '[data-testid="signature-request-scroll-button"]',
+  );
 }
 
 /**
- * This method assumes the extension is open, the dapp is open and waits for a
- * third window handle to open (the notification window). Once it does it
- * switches to the new window.
- *
+ * @deprecated since the background socket was added, and special handling is no longer necessary
+ * Just call `await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog)` instead.
  * @param {WebDriver} driver
- * @param numHandles
  */
-async function switchToNotificationWindow(driver, numHandles = 3) {
-  const windowHandles = await driver.waitUntilXWindowHandles(numHandles);
-
-  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog, windowHandles);
+async function switchToNotificationWindow(driver) {
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
```

Comment:
Can we remove this instead of deprecate it?

Reply:
Yeah that's an interesting question.  There's a second pass after this PR merges where we have to go through almost every spec file in the repo and try to eliminate most uses of `await driver.delay()`, and that's probably the time to do it.